cpp/include/cudf/detail/join.hpp

```diff
@@ -37,6 +36,10 @@
 template <typename T>
 class default_allocator;
 
+namespace cudf::structs::detail {
```

Comment:
What is the benefit of this forward declaration? Does it reduce compile time or yield some other tangible improvement?

Reply:
Forward declaration allows you to avoid the header `cudf/detail/structs/utilities.hpp` in this file.

Reply:
Not just this file, but mainly avoid including that header in `operator.cuh` which is already huge and is included in many other places.

Reply:
Thanks for that data in the other comment. Looks great. Resolving this.